src/result/completion.rs

```diff
@@ -5,57 +5,59 @@ use log::warn;
 /// problems may have been encountered along the way
 #[must_use]
 #[derive(Clone, Copy, Debug, PartialEq)]
-pub enum Completion<T> {
-    /// The operation completed without problems
-    Success(T),
-
-    /// The operation completed, but some non-fatal issues were encountered
-    Warning(T, Status),
+pub struct Completion<T> {
+    status: Status,
+    result: T,
```

Comment:
After some experience with the enum-based design, I think it hurts more than it helps, so I went back to a simpler (and slightly lighter-weight) struct.